### PR TITLE
Replace every instance of data prepper with data analyst

### LINKED ISSUE #1071: Groups: Rename `Data Prepper` to `Data Analyst`
**Description:**
The Data Prepper group needs to be renamed to accurately reflect artifacts about permissions. This includes references across the frontend and backend code which checks for a user's membership in the group.

Ideally, all group names would be defined in a single constant which could be imported across the code base so any future renaming will only have to be done in one place.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Data Prepper should be renamed to Data Analyst
- [ ] All existing and newly added permissions should still work correctly
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Update any existing migrations that add this group to use the correct group name
- [ ] Update any code references that utilize the old name to point to the new one
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**

**Supporting Documentation:**

**Open Questions:**

### PR DESCRIPTION
**Pull request by riatzukiza**
### Summary of Changes

Fixes #1071 

### How to Test

1. Check the circle CI tests
2. Check locally that the app works, if you deploy it, you will break other deployments til this is merged because of the migration, and we don't reverse the changes when we redeploy with a prior migration state.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [ ] Check against the criteria:

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

File: tdrs-backend/tdpservice/users/migrations/0015_replace_data_analyst_role.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+# Generated by Django 3.1.4 on 2021-01-08 17:41
+
```
Comment (riatzukiza) [inline tdrs-backend/tdpservice/users/migrations/0015_replace_data_analyst_role.py:3]: This was the only chunk I actually had to write anything. The rest of the changes were done through 3 different search and replaces over python files, the term showed up with 3 different spacing patterns.

File: tdrs-backend/tdpservice/users/migrations/0015_replace_data_analyst_role.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from django.contrib.auth.models import Group
+from django.db import migrations
+
+
+def change_user_groups(apps, schema_editor):
```
Comment (riatzukiza) [inline tdrs-backend/tdpservice/users/migrations/0015_replace_data_analyst_role.py:8]: We reuse the original entry for Data Prepper, just changing its name, so the users are all still connected to their original group.

**Review by abottoms-coder — APPROVED** [review]

**Review by jorgegonzalez — APPROVED** [review]
LGTM 👍

File: tdrs-backend/tdpservice/users/test/test_api/test_permissions.py [hunk 3]
```diff
@@ -26,9 +26,9 @@ def test_permission_list_not_found(api_client, create_test_users):
 
 # @pytest.mark.django_db
 # def test_permission_list_unauthorized(api_client, create_test_users):
-#     """Test data prepper does not have access."""
+#     """Test data analyst does not have access."""
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/test/test_api/test_permissions.py:29]: Not for this PR but we should definitely address these commented out tests with #888

**Review by jtwillis92 — APPROVED** [review]

**Comment by ADPennington** [comment]
test results below 👍🏾 👍🏾 

- see the change from `data prepper` to `data analyst` in groups
Screen Shot 2021-07-26 at 8 20 35 AM
Screen Shot 2021-07-26 at 8 23 06 AM
Screen Shot 2021-07-26 at 8 23 56 AM

- tested all the routes for uploading and downloading data for `data analyst` role and confirmed no change in expected behavior ✔️ 
   - can upload only clean, non-blank data files with txt or expected file extension for own STT
   - can download data for own STT only
 - tested all the routes for `ofa admin` to confirm no change in expected behavior ✔️

**Review by ADPennington — APPROVED** [review]
thanks @riatzukiza! @abottoms-coder this is ready to merge. 

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x]  Data Prepper should be renamed to Data Analyst
  - [x]  All existing and newly added permissions should still work correctly
  - [x]  Testing Checklist has been run and all tests pass

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?

### Deliverable 4: Accessible

_n/a: backend only_

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?
_tested locally instead of via deployment given note above re: migration changes. will verify upon merge into raft-tdp-main_

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_no new security issues detected_

**Pull request merged**